diff --git a/upstream/emscripten/tools/emscripten.py b/upstream/emscripten/tools/emscripten.py
index f73e2e1..9129d4f 100644
--- a/upstream/emscripten/tools/emscripten.py
+++ b/upstream/emscripten/tools/emscripten.py
@@ -627,7 +627,7 @@ def create_tsd_exported_runtime_methods(metadata):
     tsc = ['tsc']
   else:
     tsc = shared.get_npm_cmd('tsc')
-  cmd = tsc + ['--outFile', tsc_output_file, '--declaration', '--emitDeclarationOnly', '--allowJs', js_doc_file]
+  cmd = tsc + ['--outFile', tsc_output_file, '--declaration', '--skipLibCheck', '--emitDeclarationOnly', '--allowJs', js_doc_file]
   shared.check_call(cmd, cwd=path_from_root())
   return utils.read_file(tsc_output_file)
 
